python_utils/regular_to_cube/fv3_cold_starts.py

```diff
@@ -0,0 +1,568 @@
+            else:
+                # Create a level coordinate if it doesn't exist
+                lev_coord = 'pfull'
+                ds_out[lev_coord] = (lev_coord, np.arange(1, nlev + 1))
+                ds_out[lev_coord].attrs = {'units': 'level', 'long_name': 'pressure level'}
+            
+            if 'grid_yt' in ds.coords and 'grid_xt' in ds.coords:
+                lat_coord, lon_coord = 'grid_yt', 'grid_xt'
+            elif 'lat' in ds.coords and 'lon' in ds.coords:
+                lat_coord, lon_coord = 'lat', 'lon'
+            else:
+                # Use existing coordinate names from a 3D variable
+                for var in ds.data_vars:
+                    if len(ds[var].dims) == 3:
+                        lat_coord, lon_coord = ds[var].dims[1], ds[var].dims[2]
+                        break
+                else:
+                    raise ValueError("Cannot determine coordinate names")
+            
+            # Create the DataArray for the field
+            field_da = xr.DataArray(
+                field_array,
+                dims=[lev_coord, lat_coord, lon_coord],
+                coords={
+                    lev_coord: ds_out[lev_coord],
+                    lat_coord: ds_out[lat_coord],
+                    lon_coord: ds_out[lon_coord]
+                },
+                name=field_name
+            )
+            
+            # Add metadata if provided
+            if field_metadata and field_name in field_metadata:
+                field_da.attrs.update(field_metadata[field_name])
+            else:
+                # Default metadata
+                field_da.attrs = {
+                    'units': 'kg kg-1',
+                    'long_name': f'{field_name} mass mixing ratio'
+                }
+            
+            # Add the field to the dataset
+            ds_out[field_name] = field_da
+            
+            logger.info(f"Added {field_name}: shape {field_array.shape}, "
+                       f"min={np.min(field_array):.2e}, max={np.max(field_array):.2e}")
+    
+    # Determine output path
+    if output_path is None:
+        output_path = f"{tile_file_path}.out.nc"
+        # Create backup if requested
+        if backup_original:
+            backup_path = f"{tile_file_path}.backup"
+            if not Path(backup_path).exists():
+                logger.info(f"Creating backup: {backup_path}")
+                shutil.copy2(tile_file_path, backup_path)
+    
+    # Write the modified dataset
+    logger.info(f"Writing modified dataset to: {output_path}")
+    ds_out.to_netcdf(output_path)
+
+    logger.info(f"Successfully added {len(field_data)} fields to {output_path}")
+        
+    # Move the temporary file to the final output path
+    if output_path != tile_file_path:
+        shutil.move(output_path, tile_file_path)
+        logger.info(f"Moved temporary output file to: {tile_file_path}")
+
+    return output_path
+
+
+def create_aerosol_field_metadata() -> Dict[str, Dict]:
```

Comment:
Like the use of netcdf attributes.  wonder if (in the future not this PR) if we have this be configurable through either a toml or yaml file.  toml is officially supported in python 3.11+ without the need for additional libraries but pyyaml is already heavily used.

Reply:
I like that idea, but would say YAML since `wxflow` has a lot of YAML and Jinja support already

Reply:
Fine with either but trying to think ahead to make this more flexible for whatever our needs are.  We can add it to the infamous TODO list